lib/storage/index.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * On Web and Desktop it's enough to leave everything to localforage
+ */
+
+import Storage from './providers/LocalForage';
+
+const webStorage = {
+    ...Storage,
+
+    /**
+     * Storage synchronization mechanism keeping all opened tabs in sync
+     * @param {object} config
+     * @param {function(key: string, data: any)} config.keyChanged
+     */
+    synchronizeInstances(config) {
+        const SYNC_ONYX = 'SYNC_ONYX';
+
+        // Override `setItem` to raise storage events that we can intercept in other tabs
+        this.setItem = (key, value) => Storage.setItem(key, value)
+            .then(() => {
+                /** We make a temp write to storage just to raise the storage event (below)
+                 * then remove the "dummy" value.
+                 * Listeners would receive everything they need to read the new value from the DB */
+                global.localStorage.setItem(SYNC_ONYX, key);
+                global.localStorage.removeItem(SYNC_ONYX);
+            });
+
+        // This event will be triggered on tabs other than the current tab
+        global.addEventListener('storage', (event) => {
+            if (!event.newValue) {
```

Comment:
Yeah I try to explain this with the added code comments 

```

Reply:
So you're question is whether this would still raise the two events and raise them in the correct order 

```js
global.localStorage.setItem(SYNC_ONYX, key);
global.localStorage.removeItem(SYNC_ONYX);
```

Yes it will
It will first raise an event with something like `{ key: 'SYNC_ONYX', value: 'session' }` 
And immediately after `{ key: 'SYNC_ONYX', value: null }`   
It won't directly skip to the `null` case or fire them in reverse order